/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
==============================================================================*/

syntax = "proto2";

package lstm_object_detection.tflite.protos;



// This is dervice from TensorFlow's SsdAnchorGenerator proto that is used to
// configures TensorFlow's anchor generator.
// object_detection/protos/ssd_anchor_generator.proto
message AnchorGenerationOptions {
  // The input image width in pixels
  optional int32 image_width = 1;

  // The input image height in pixels
  optional int32 image_height = 2;

  // The base anchor width in pixels
  optional int32 base_anchor_width = 3;

  // The base anchor height in pixels
  optional int32 base_anchor_height = 4;

  // The minimum anchor scaling (should be < 1.0)
  optional float min_anchor_scale = 5;

  // The maximum anchor scaling
  optional float max_anchor_scale = 6;

  // List of aspect ratios to generate anchors for. Aspect ratio is specified as
  // (width/height)
  repeated float anchor_aspect_ratios = 7 [packed = true];

  // List of strides in pixels for each layer
  repeated int32 anchor_strides = 8 [packed = true];

  // List of offset in pixels for each layer
  repeated int32 anchor_offsets = 9 [packed = true];
}
